### ISSUE TITLE
Add option to filter master server output

### ISSUE THREAD

**Issue by ChillerDragon** [teeworlds/teeworlds#2442]
When a teeworlds server is running without players for some time it is really annoying to read the logs since they are spammed with master server information:

```
[...]
[2020-01-28 10:01:58][register]: fetching server counts
[2020-01-28 10:01:58][register]: chose 'master2.teeworlds.com' as master, sending heartbeats
[2020-01-28 10:01:59][register]: no firewall/nat problems detected
[2020-01-28 10:01:59][register]: server registered
[2020-01-28 10:32:00][engine/mastersrv]: refreshing master server addresses
[2020-01-28 10:32:00][register]: refreshing ip addresses
[2020-01-28 10:32:05][engine/mastersrv]: saving addresses
[2020-01-28 10:32:05][register]: fetching server counts
[2020-01-28 10:32:05][register]: chose 'master2.teeworlds.com' as master, sending heartbeats
[2020-01-28 10:32:09][register]: no firewall/nat problems detected
[2020-01-28 10:32:09][register]: server registered
[2020-01-28 11:02:10][engine/mastersrv]: refreshing master server addresses
[2020-01-28 11:02:10][register]: refreshing ip addresses
[2020-01-28 11:02:15][engine/mastersrv]: saving addresses
[...]
```

So it would either be nice if there would be a config to turn off all master server messages including error messages.

Or do only show the success messages once or if there was a error previously.

The second version still causes spam when the master servers are down tho. So I would prefer both solutions at once.

**Comment by Dune-jr** [comment]
I think the categories are there to help filtering what you want to see. You can `|grep -v register`.

**Comment by ChillerDragon** [comment]
@Dune-jr yes i know but i am sick of piping my logs through tons of greps before i can take a quick glance at what happend. Also your grep would also filter every other message that im interested in that might contain ``register``.

**Comment by jxsl13** [comment]
I would suggest to only be able to disable the visibility of successful heartbeats, but errors should always be visible.

**Comment by ChillerDragon** [comment]
@jxsl13 remember the big 0.6 master outage? My logs were a mess.

**Comment by Robyt3** [comment]
I think IConsole could have more different and intuitive logging severities.

```
OUTPUT_LEVEL_STANDARD=0,
OUTPUT_LEVEL_ADDINFO,
OUTPUT_LEVEL_DEBUG,
```

I propose to at least differenciate between the first 4 of these:

- ERROR
- WARNING
- INFO
- DEBUG
- FATAL (severe errors that will lead to termination)
- TRACE (even more detailed debug output)

The console could also support filtering based on categories, by specifying excluded categories in a config var.

**Comment by oy** [comment]
The levels are just about the ingame console, stdout or the logfile should show them all.

**Comment by oy** [comment]
Could probably shrink these messages to 1 per master register and show the rest just in debug?

**Comment by jxsl13** [comment]
one message per state change.

**Comment by Dune-jr** [comment]
@oy sounds good to me
